src/main/java/org/testng/internal/Systematiser.java

```diff
@@ -0,0 +1,86 @@
+package org.testng.internal;
+
+import org.testng.ITestNGMethod;
+
+import java.util.Comparator;
+
+public class Systematiser {
+
+    private Systematiser() {
+        //Utility class. Defeat instantiation.
+    }
+
+    public static Comparator<Graph.Node> getComparator() {
```

Comment:
@juherr - Ok, but I would like to understand why does `Node` object have to be tied to `ITestNGMethod` ? There are tests for the `Graph` class viz., [GraphTest](https://github.com/cbeust/testng/blob/master/src/test/java/test/GraphTest.java#L21-L27) which attempts at testing `Graph` behavior using Strings. If we basically tie down the `Comparator` to `ITestNGMethod` then this would cause changes to be required in the test as well and by behavior one shouldn't have to need an `ITestNGMethod` object to test the behavior of the `Graph`. That is why I didn't change it.